tests/users/test_dao_update_user.py

```diff
@@ -1,37 +1,111 @@
+        self.assertEqual("good_social_media_links", self.admin_user.social_media_links)
+        self.assertEqual("good_skills", self.admin_user.skills)
+        self.assertEqual("good_interests", self.admin_user.interests)
+        self.assertEqual("good_resume_url", self.admin_user.resume_url)
+        self.assertEqual("good_photo_url", self.admin_user.photo_url)
 
     def test_update_fields_with_empty_data(self):
 
+        name_before_update = self.admin_user.name
+        self.assertIsNotNone(self.admin_user.name)
+
+        self.assertIsNone(self.admin_user.bio)
+        self.assertIsNone(self.admin_user.location)
         self.assertIsNone(self.admin_user.occupation)
         self.assertIsNone(self.admin_user.organization)
+        self.assertIsNone(self.admin_user.slack_username)
+        self.assertIsNone(self.admin_user.social_media_links)
+        self.assertIsNone(self.admin_user.skills)
+        self.assertIsNone(self.admin_user.interests)
+        self.assertIsNone(self.admin_user.resume_url)
+        self.assertIsNone(self.admin_user.photo_url)
 
-        data = dict(occupation="good_developer", organization="good_org")
+        data = dict(
+            name="",
+            bio="",
+            location="",
+            occupation="",
+            organization="",
+            slack_username="",
+            social_media_links="",
+            skills="",
+            interests="",
+            resume_url="",
+            photo_url="",
+        )
         UserDAO.update_user_profile(self.admin_user.id, data)
 
-        self.assertEqual("good_developer", self.admin_user.occupation)
-        self.assertEqual("good_org", self.admin_user.organization)
-
-        data = dict(occupation="", organization="")
-        UserDAO.update_user_profile(self.admin_user.id, data)
+        name_after_update = self.admin_user.name
+        self.assertIsNotNone(self.admin_user.name)
+        self.assertEqual(name_before_update, name_after_update)
 
+        self.assertIsNone(self.admin_user.bio)
+        self.assertIsNone(self.admin_user.location)
         self.assertIsNone(self.admin_user.occupation)
         self.assertIsNone(self.admin_user.organization)
+        self.assertIsNone(self.admin_user.slack_username)
+        self.assertIsNone(self.admin_user.social_media_links)
+        self.assertIsNone(self.admin_user.skills)
+        self.assertIsNone(self.admin_user.interests)
+        self.assertIsNone(self.admin_user.resume_url)
+        self.assertIsNone(self.admin_user.photo_url)
+
+    def test_update_user_that_does_not_exist(self):
+
+        user = UserModel.query.filter_by(id=2).first()
+        self.assertIsNone(user)
+
+        data = dict(occupation="good_developer", organization="good_org")
+        dao_result = UserDAO.update_user_profile(user, data)
+
+        self.assertEqual(
+            (messages.USER_DOES_NOT_EXIST, HTTPStatus.NOT_FOUND), dao_result
```

Comment:
![ezgif com-gif-maker (71)](https://user-images.githubusercontent.com/29667122/123967181-6c2e7480-d9f9-11eb-85bb-0916e4765e3b.gif)

Reply:
BTW, the decorator is used on this function
<img width="549" alt="Screen Shot 2021-06-30 at 11 07 50 pm" src="https://user-images.githubusercontent.com/29667122/123968830-f4614980-d9fa-11eb-865c-cba27a38f254.png">